lib/fluent/plugin/buffer.rb

```diff
@@ -761,20 +771,25 @@ def write_step_by_step(metadata, data, format, splits_count, &block)
             begin
               while writing_splits_index < splits.size
                 split = splits[writing_splits_index]
+                formatted_split = format ? format.call(split) : split.first
+                if split.size == 1 && original_bytesize == 0
+                  big_record_size = formatted_split.bytesize
+                  if chunk.bytesize + big_record_size > @chunk_limit_size
+                    errors << "a #{big_record_size} bytes record (nth: #{writing_splits_index}) is larger than buffer chunk limit size"
+                    writing_splits_index += 1
+                    next
+                  end
```

Comment:
It's OK for the case that `format` is given.
But the case on `format == nil` and `@compress == :gzip`, `BufferChunkOverflowError` might not occurred in actual.
In such case, doing `chunk.commit` here and delaying to store errors  until `if chunk_size_over?(chunk)` block is better like you did in the previous patch,

Reply:
> But the case on `format == nil` and `@compress == :gzip`, `BufferChunkOverflowError` might not occurred in actual. 

https://github.com/fluent/fluentd/blob/46dad90aac2d463bf3d5b0d1be372ac97c0d0724/lib/fluent/plugin/buffer/chunk.rb#L187-L199

Reply:
okay, I'll fix  it.

Reply:
Fixed.